lib/libc/common/Kconfig

```diff
@@ -66,3 +66,11 @@ config COMMON_LIBC_STRNLEN
 	bool
 	help
 	  common implementation of strnlen().
+
+config COMMON_LIBC_THRD
+	bool "C11 <threads.h> API support"
+	depends on DYNAMIC_THREAD
+	depends on POSIX_API
```

Comment:
Worth add a comment here that the POSIX_API `depends` is only because its implemented utilizing that and could be re-implemented using native zephyr api's